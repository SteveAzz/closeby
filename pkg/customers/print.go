package customers

import (
	"fmt"
	"io"
	"sort"
	"text/tabwriter"
)

// PrintCustomers write to io.Writer with the help of a tab write to format the
// given customers. Only show the customer ID and the name, which is sorted by
// ID in ascending order.
func PrintCustomers(w io.Writer, customers []Customer) error {
	sort.Sort(ByID(customers))

	fmt.Fprint(w, "\tCLOSE BY CUSTOMERS\n\n")
	tab := tabwriter.NewWriter(w, 0, 2, 2, ' ', 0)
	for _, c := range customers {
		fmt.Fprintf(w, "\t%d \t%s\n", c.ID, c.Name)
	}
	fmt.Fprint(w, "\n")

	return tab.Flush()
}
