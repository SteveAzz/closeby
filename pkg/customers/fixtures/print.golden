	CLOSE BY CUSTOMERS

	1 	Alice Cahill
	8 	Eoin Ahearn
	12 	Christina McArdle

